compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -110,8 +42,77 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
     ) -> InterpResult<'tcx, bool> {
         let instance_args = instance.args;
         let intrinsic_name = self.tcx.item_name(instance.def_id());
+        let tcx = self.tcx.tcx;
 
         match intrinsic_name {
+            sym::type_name => {
```

Comment:
nit: I see this is a code-move & fixup to work in here, but is it worth sorting these arms into the rest of the match? though I guess the other arms are also not sorted simply... it was mostly that `needs_drop` is between `type_id` & `type_name` that caught my eye hehe

Reply:
yea, could do that, good point. didn't want to do it in the first commit to kinda make the diff line up